src/utilstrencodings.cpp

```diff
@@ -538,3 +538,118 @@ int atoi(const std::string& str)
 {
     return atoi(str.c_str());
 }
+
+/** Upper bound for mantissa.
+ * 10^18-1, largest arbitrary decimal that will fit in 64 bit.
```

Comment:
Hypernit: this comment seems inaccurate. There are definitely larger integers that fit in 64 bit.